### PR TITLE
Issue26

### PR DESCRIPTION
**Pull request by matthewgardner**
Enhancement/Issue26 - Allow user to state filter.mode=named and only process mac addresses listed in ruuvi-names.properties. If filter.mode is set to named but nothing populated in  ruuvi-names.properties an exception will be thrown the collector will terminate with error.

### CODE REVIEW

File: src/test/java/fi/tkgwf/ruuvi/config/ConfigTest.java [hunk 1]
```diff
@@ -170,4 +170,26 @@ void testInfluxDbFieldFilter() {
...
+        assertTrue(Config.isAllowedMAC("XX12CD34EF56"));
+        assertTrue(Config.isAllowedMAC("ABCDEFG"));
+        assertFalse(Config.isAllowedMAC(null));
+        
+        final Properties properties = new Properties();
+        properties.put("filter.mode", "named");
+        try {
+            Config.readConfigFromProperties(properties);
+        } catch (final IllegalStateException expected) {
+            fail("There should have been an exception: ruuvi-names.properties is empty.");
```
Comment (Scrin) [inline src/test/java/fi/tkgwf/ruuvi/config/ConfigTest.java]: isn't this wrong?

Comment (matthewgardner) [reply src/test/java/fi/tkgwf/ruuvi/config/ConfigTest.java]: yes - it should say it shouldnt have been an exception.

Comment (matthewgardner) [reply src/test/java/fi/tkgwf/ruuvi/config/ConfigTest.java]: in fact that try catch is misleading - ill remove it

Comment (matthewgardner) [reply src/test/java/fi/tkgwf/ruuvi/config/ConfigTest.java]: unless we want to refactor and make it possible to change the contents of TAG_NAMES ?

Comment (Scrin) [reply src/test/java/fi/tkgwf/ruuvi/config/ConfigTest.java]: Better test coverage is always better, but not strictly required. You can do the refactoring and add testing for that, if you want

**Pull request closed without merging**